### DISCUSSION TITLE
flaml crashes on MacOS Monterey v 12.3.1

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/FLAML#543] [General]
Call to resource.setrlimit(resource.RLIMIT_AS, (memory_limit, hard)) causes error
<img width="1399" alt="Screen Shot 2022-05-05 flaml crash" src="https://user-images.githubusercontent.com/90455225/167453259-0e30f323-0ae6-46ae-ab4d-2b20b3900006.png">
:

**Comment by sonichi** [comment]
I created an issue #547 and proposed a change in the code. Please feel free to create a PR or suggest other solutions.
